### ISSUE TITLE
Python 3.10 - threading.Thread - DeprecationWarnings: setDaemon() and setName()

### ISSUE THREAD

**Issue by slieberth** [ncclient/ncclient#552]
I got two deprecation warnings, when testing with python=3.10 in ubuntu 22.4: 

```
  /usr/local/lib/python3.10/dist-packages/ncclient/transport/session.py:44: DeprecationWarning: setDaemon() is deprecated, set the daemon attribute instead
    self.setDaemon(True)
```
should be changed to:
```
    self.deamon = True
```
and 
```
  /usr/local/lib/python3.10/dist-packages/ncclient/transport/session.py:47: DeprecationWarning: setName() is deprecated, set the name attribute instead
    self.setName('session')
```
should be changed to:
```
    self.name = 'session'
```

if I shall create a pull/merge request for those changes, please let me know

best regards Stefan

**Comment by einarnn** [comment]
@slieberth Yes, please submit a PR, thanks! Please consider that we would like maintain backwards compatibility to Python 3.5. And please consider adding 3.10 to the test matrix in [check.yaml](https://github.com/ncclient/ncclient/tree/master/.github/workflows/check.yaml)

**Comment by einarnn** [comment]
This has been addressed in #553
